pkg/package-server/provider/registry.go

```diff
@@ -165,9 +200,107 @@ func (p *RegistryProvider) syncCatalogSource(obj interface{}) (syncError error)
 	p.setClient(newRegistryClient(source, conn), key)
 	logger.Info("new grpc connection added")
 
+	syncError = p.refreshCache(context.Background(), key)
```

Comment:
This is my only real design comment - we don't need to rebuild the cache on every sync. We should only need to do it when the catalogsource changes, which should be indicated by the lastupdatetime.